Plugins/GenerateDoccReference/GenerateDoccReference.swift

```diff
@@ -18,83 +21,101 @@ struct GenerateDoccReferencePlugin: CommandPlugin {
     context: PluginContext,
     arguments: [String]
   ) async throws {
-    // Locate generation tool.
     let generationTool = "generate-docc-reference"
     let generationToolFile = try context.tool(named: generationTool).path
-
-    // Create an extractor to extract plugin-only arguments from the `arguments`
-    // array.
     var extractor = ArgumentExtractor(arguments)
 
-    // Run generation tool once if help is requested.
-    if extractor.helpRequest() {
-      try generationToolFile.exec(arguments: ["--help"])
-      print(
-        """
-        ADDITIONAL OPTIONS:
-          --configuration <configuration>
-                      Tool build configuration used to generate the
-                      reference document. (default: release)
-
-        NOTE: The "GenerateDoccReference" plugin handles passing the "<tool>" and
-        "--output-directory <output-directory>" arguments. Manually supplying
-        these arguments will result in a runtime failure.
-        """)
+    if checkForHelpArgument(extractor: &extractor, generationToolFile: generationToolFile) {
       return
     }
 
-    // Extract configuration argument before making it to the
-    // "generate-docc-reference" tool.
     let configuration = try extractor.configuration()
+    let buildArtifacts = try buildPackage(configuration: configuration)
+
+    try generateDocs(context: context, with: extractor, buildArtifacts: buildArtifacts, generationToolFile: generationToolFile)
+  }
 
-    // Build all products first.
-    print("Building package in \(configuration) mode...")
+  private func buildPackage(configuration: PackageManager.BuildConfiguration) throws -> [PackageManager.BuildResult.BuiltArtifact] {
+    // The package must be built before documentation can be generated
+    // because DocC requires compiled executables to extract symbols.
     let buildResult = try packageManager.build(
       .all(includingTests: false),
       parameters: .init(configuration: configuration))
 
+    // If the build fails, documentation generation cannot proceed.
     guard buildResult.succeeded else {
       throw GenerateDoccReferencePluginError.buildFailed(buildResult.logText)
     }
+
     print("Built package in \(configuration) mode")
+    return buildResult.builtArtifacts
+  }
 
-    // Run generate-docc-reference on all executable artifacts.
-    for builtArtifact in buildResult.builtArtifacts {
-      // Skip non-executable targets
+  private func generateDocs(
+    context: PluginContext, with extractor: ArgumentExtractor,
+    buildArtifacts: [PackageManager.BuildResult.BuiltArtifact],
+    generationToolFile: Path
+  ) throws {
+    for builtArtifact in buildArtifacts {
+      // DocC documentation is only relevant for executable targets,
+      // since documentation is typically generated for command-line tools.
```

Comment:
This comment is slightly confusing and misleading. DocC documentation extends well beyond executable targets. 

It would be more correct to say that this plugin is only relevant for common-line tools, which are executables.